text_2_sql/text_2_sql_core/src/text_2_sql_core/prompts/sql_schema_selection_agent.yaml

```diff
@@ -5,39 +5,67 @@ description:
-    The user's question will be related to {{ use_case }}.
-  </scope_of_user_query>
-
   <instructions>
-    1. Extract key terms, filter conditions, and entities from the user's question.
-    2. Perform entity recognition on these key terms to identify categories they might belong to.
-    3. Extract filter conditions that are string representations. Exclude numerical or date values.
-    4. Expand acronyms or abbreviations in the user's question to their full forms alongside the acronyms.
-  </instructions>
-
-  <example>
-    Show me the list of employees in the HR department employed during 2008?
+    1. Extract key terms, filter conditions, and entities from the user's question
+    2. Group related entities that might need to be joined together
+    3. Identify all potential filter conditions, including:
+       - Geographic terms (countries, regions, cities)
+       - Temporal terms (dates, months, years)
+       - Product categories and attributes
+       - Customer segments
+    4. Consider relationship paths between entities
+    5. Expand acronyms or abbreviations alongside their original form
 
-    Entities & Key Terms:
-      employees, HR department, year
-
-    Entities & Key Terms Groups:
-      [[\"people\", \"employees\"], [\"departments\", \"teams\"], [\"date\", \"year\"]]
+    Important: When dealing with sales analysis:
+    - Always consider both the transaction tables (e.g., SalesOrderHeader) and their related dimension tables
+    - For geographic queries, include both address and customer-related tables
+    - For temporal queries, identify tables with date/time columns
+    - For product queries, include category and description tables
+  </instructions>
 
-    Filter Conditions:
-      HR, HR Department, Human Resources, Human Resources Department
+  <examples>
+    Example 1: \"What country did we sell the most to in June 2008?\"
+    {
+      \"entities\": [
+        [\"sales\", \"orders\", \"transactions\"],
+        [\"country\", \"region\", \"geography\", \"location\"],
+        [\"customer\", \"buyer\", \"client\"],
+        [\"address\", \"shipping\", \"destination\"]
+      ],
+      \"filter_conditions\": [
+        \"country\", \"region\"
+      ]
+    }
 
-    Exclude numerical values like 2008, as it is a DateTime value.
-  </example>
+    Example 2: \"What are the total sales for mountain bikes in 2008?\"
+    {
+      \"entities\": [
+        [\"sales\", \"orders\", \"transactions\"],
+        [\"product\", \"item\", \"merchandise\"],
+        [\"category\", \"type\", \"classification\"],
+        [\"bike\", \"bicycle\", \"cycle\"]
+      ],
+      \"filter_conditions\": [
+        \"mountain\", \"mountain bike\", \"bike\"
+      ]
+    }
+  </examples>
 
   <output-format>
     {
       \"entities\": [[\"<entity_1>\", \"<entity_2>\"], [\"<entity_3>\"]],
       \"filter_conditions\": [\"<filter_condition_1>\", \"<filter_condition_2>\"]
     }
   </output-format>
+
+  <key_relationships>
```

Comment:
Not sure we can put this in as its specific to the dataset? Can we make it a parameter if it is needed?